### PR TITLE
[AArch64] Implement intrinsics for SME FP8 FMLAL/FMLALL (Indexed)

### PR DESCRIPTION
**Pull request by SpencerAbson**
This patch implements the following intrinsics:

Multi-vector 8-bit floating-point multiply-add long.
``` c
  // Only if __ARM_FEATURE_SME_F8F16 != 0
  void svmla_lane_za16[_mf8]_vg2x1_fpm(uint32_t slice, svmfloat8_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx,
                                       fpm_t fpm)  __arm_streaming __arm_inout("za");

  void svmla_lane_za16[_mf8]_vg2x2_fpm(uint32_t slice, svmfloat8x2_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx,
                                       fpm_t fpm)  __arm_streaming __arm_inout("za");

  void svmla_lane_za16[_mf8]_vg2x4_fpm(uint32_t slice, svmfloat8x4_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx
                                       fpm_t fpm) __arm_streaming __arm_inout("za");

// Only if __ARM_FEATURE_SME_F8F32 != 0
  void svmla_lane_za32[_mf8]_vg4x1_fpm(uint32_t slice, svmfloat8_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx,
                                       fpm_t fpm)__arm_streaming __arm_inout("za");

  void svmla_lane_za32[_mf8]_vg4x2_fpm(uint32_t slice, svmfloat8x2_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx,
                                       fpm_t fpm)__arm_streaming __arm_inout("za");

  void svmla_lane_za32[_mf8]_vg4x4_fpm(uint32_t slice, svmfloat8x4_t zn,
                                       svmfloat8_t zm, uint64_t imm_idx,
                                       fpm_t fpm)__arm_streaming __arm_inout("za");
```
In accordance with: https://github.com/ARM-software/acle/pull/323

### CODE REVIEW

File: llvm/lib/Target/AArch64/SMEInstrFormats.td [hunk 1]
```diff
@@ -1965,21 +1987,91 @@ class sme2_mla_long_array_index_base<bits<2> op0, bits<2> op, Operand index_ty,
...
-  let Inst{4-3}   = op;
 
   let Constraints = "$ZAda = $_ZAda";
 }
 
-multiclass sme2_mla_long_array_index<string mnemonic, bits<2> op0, bits<2> op, ValueType zpr_ty, SDPatternOperator intrinsic> {
-  def _HtoS : sme2_mla_long_array_index_base<op0, op, uimm3s2range, ZPR16,
-                                          mnemonic>, SMEPseudo2Instr<NAME # _HtoS, 1> {
-    bits<3> i3;
+multiclass sme2_mla_long_array_index_za16<string mnemonic, SDPatternOperator intrinsic> {
```
Comment (CarolineConcatto) [inline llvm/lib/Target/AArch64/SMEInstrFormats.td:1994]: I am not sure why you did the refactoring this way.
It is possible to review, but it is quite confusing to check. I would not like to trust only in the tests.
Is it possible to no mix the refactoring with the intrinsic implementation?

Comment (SpencerAbson) [reply llvm/lib/Target/AArch64/SMEInstrFormats.td:1994]: That's fine, I'll save it for a another time :+1: .

**Comment by CarolineConcatto** [comment]
I will wait you merge the previous patch to have a look on this one again later!

**Review by CarolineConcatto — APPROVED** [review]
Thank you Spencer.
LGTM!

**Pull request merged**